### DISCUSSION TITLE
Can Puppet be solved in 1 transaction ?

### DISCUSSION THREAD

**Discussion by MiniGlome** [tinchoabbate/damn-vulnerable-defi#26] [Help / Q&A / Troubleshooting] [answered]
Hello,
The new release of the Puppet Challenge requires the player to use only 1 transaction to succeed. However, I don't see how this is possible since in order to solve the challenge, the player has to deploy a smart contract and approve the token. That's at least 2 transactions...

**Comment by naiba** [comment]
It is feasible and I have done it.

**Reply by devival** [reply]
Any hints? how?

**Comment by naiba** [comment]
I suggest you spend some more time learning and trying. There is a flaw in
the thirteenth challenge that I only discovered after watching it for five
nights.

**Comment by naiba** [comment]
I can only assure you that this challenge is achievable.

**Comment by tinchoabbate** [comment, marked as answer]
Hey @MiniGlome, as @naiba was saying, it's possible. Keep looking ! 💪

**Reply by Aamirusmani1552** [reply]
Do you know a way to make the transaction signature on-chain using solidity.
I think I might have an idea to solve this.
Not able to find a way so that we can generate the signature on chain. Or maybe it's not possible. I don't know.
can you help to find this @MiniGlome  @naiba

**Reply by Aamirusmani1552** [reply]
Finally solved the problem.
Yes it is achievable

**Comment by pendor-lifeform** [comment]
Figuring out how to do it in a single transaction took me more time than the actual bug.
There are multiple "solutions" online, that simply deploy contract with eth already - that's not THE solution. 🤐
